solidity/ecdsa/deploy/12_upgrade_random_beacon.ts

```diff
@@ -0,0 +1,26 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+import type { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async (hre: HardhatRuntimeEnvironment) => {
+  const { getNamedAccounts, deployments } = hre
+  const { governance } = await getNamedAccounts()
+  const { execute } = deployments
+
+  const RandomBeacon = await deployments.get("RandomBeacon")
+
+  await execute(
+    "WalletRegistryGovernance",
+    { from: governance, log: true, waitConfirmations: 1 },
+    "upgradeRandomBeacon",
+    RandomBeacon.address
+  )
+}
```

Comment:
Why do we do it?

Reply:
I did it because some unit tests fail if we don't use the real `RandomBeacon`.

Reply:
It makes sense, although it also means that the real `RandomBeacon` will be needed during the chaosnet deployment - because it will have to be set in the `WalletRegistry`'s constructor.

Reply:
I think it makes sense because right now it gets pretty complicated and I must confess I was confused looking at deployment scripts and trying to grasp the dependency graph.

The original deployment was with a real `RandomBeacon`. The final version of the deployment will be with a real `RandomBeacon`. All the `solidity/ecdsa` unit tests were either implemented for a real `RandomBeacon` or for the mocked instance.

If we replace the dependency for chaosnet deployments to `RandomBeaconChaosnet` in one final step, then we are doing the minimal number of modifications needed and we can easily unplug it later. It's not the most efficient approach in terms of deployment time but I think we can sacrifice it for less complexity. It also mimicks what we are going to do on mainnet.

To sum up, I think we need:
- The resolve script `00_resolve_random_beacon.ts`, always triggered.
- The resolve script `00_resolve_random_beacon_chaosnet.ts`, triggered only for chaosnet
- The upgrade script, `99_upgrade_random_beacon_chaosnet.ts`, triggered only for chaosnet

Additionally, we will tag as chaosnet the following networks in `hardhat.config.ts`:
- `goerli`
- `development`

Does it make sense?

I would also explain in a comment in `99_upgrade_random_beacon_chaosnet.ts` our reasoning - why we decided for an upgrade for chaosnet.

Reply:
Done in b7035c52994d1179c98bcb46d84a4fb7d5da8a8d.

However, I have a few thoughts:
1) Shouldn't `mainnet` network be marked as chaosnet since the next deployment for mainnet should include `RandomBeaconChaosnet`?

2) Should we introduce the `chaosnet` tag to random-beacon deployments? The [script that deploys RandomBeaconChasonet](https://github.com/keep-network/keep-core/blob/a0817e53bc6ce1c85013ee254d24b88cc78b881a/solidity/random-beacon/deploy/09_deploy_random_beacon_chaosnet.ts#L14) could be run only for chasonet deployments.